src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1085,4 +1088,135 @@ BOOST_AUTO_TEST_CASE(CustomRecursiveReberGrammarTest)
   ReberGrammarTestCustomNetwork(16, true);
 }
 
+/**
+ * @brief Generates noisy sine wave and outputs the data and the labels that
+ * can be used directly for training and testing with RNN.
+ *
+ * @param data    The data points as output
+ * @param labels  The expected values as output
+ * @param rho     The size of the sequence of each data point
+ * @param outputSteps How many output steps to consider for every rho inputs
+ * @param dataPoints  The number of generated data points. The actual generated
+ * data points may be more than this to adjust to the outputSteps.
+ * But at the minimum these many data points will be generated.
+ * @param gain    The gain on the amplitude
+ * @param freq    The frquency of the sine wave
+ * @param phase   The phase shift if any
+ * @param noisePercent  The percent noise to induce
+ * @param numCycles  How many full size wave cycles required. All the data
+ * points will be fit into these cycles.
+ * @param normalize Whether to normalise the data. This may be required for some
+ * layers like LSTM. Default is true.
+ */
+void GenerateNoisySinRNN(arma::cube& data, arma::cube& labels, size_t rho,
+                         size_t outputSteps = 1, const int dataPoints = 100,
```

Comment:
Can you use a single line for each parameter? See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#method-declarations for more information, I think right now this isn't easy to read.